lightning/src/ln/channelmanager.rs

```diff
@@ -556,6 +558,27 @@ pub(super) struct PeerState<Signer: ChannelSigner> {
 	/// been assigned a `channel_id`, the entry in this map is removed and one is created in
 	/// `channel_by_id`.
 	pub(super) inbound_v1_channel_by_id: HashMap<[u8; 32], InboundV1Channel<Signer>>,
+	/// `(temporary_)channel_id` -> `OutboundV2Channel`.
+	///
+	/// Holds all outbound V2 channels where the peer is the counterparty. V2 channels are assigned
+	/// a `channel_id` before a funding transaction is created interactively as it's derived from
+	/// both parties' revocation basepoints once these are known. Hence, this map's keys are either
+	/// temporary channel IDs or channel IDs.
+	///
+	/// The entries in this map are only moved to `channel_by_id` once interactive transaction
+	/// construction completes successfully.
+	pub(super) outbound_v2_channel_by_id: HashMap<[u8; 32], OutboundV2Channel<Signer>>,
```

Comment:
Note if we take more memory state for dual-funding than legacy open flow, we might need to revisit the limits we introduced with #1889.

Reply:
Thanks will investigate this 👍